llvm/include/llvm/ADT/APFloat.h

```diff
@@ -322,7 +322,38 @@ struct APFloatBase {
 
 namespace detail {
 
-class IEEEFloat final : public APFloatBase {
```

Comment:
Is there any reason to keep APFloatBase as a class? It seems like it's just a namespace

Reply:
I tried this before submitting the patch. We cannot import an unscoped enum from a namespace to a class.

Reply:
Most of these enums should probably move to enum class anyway

Reply:
These unscoped enums are widely used in llvm via `APFloat::***`. It would be a heavy work to convert all uses of unscoped enums to scoped enums :(